src/java/org/apache/cassandra/index/sai/disk/v1/segment/SegmentRamBuffer.java

```diff
@@ -15,34 +15,34 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.cassandra.index.sai.disk.v1.bbtree;
+package org.apache.cassandra.index.sai.disk.v1.segment;
 
 import java.util.concurrent.atomic.LongAdder;
 import javax.annotation.concurrent.NotThreadSafe;
 
 import org.apache.cassandra.db.memtable.TrieMemtable;
 import org.apache.cassandra.db.tries.InMemoryTrie;
 import org.apache.cassandra.utils.Throwables;
-import org.apache.cassandra.utils.bytecomparable.ByteSource;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
 import org.apache.lucene.util.packed.PackedInts;
 import org.apache.lucene.util.packed.PackedLongValues;
 
 /**
- * On-heap buffer for values that provides a sorted view of itself as a {@link BlockBalancedTreeIterator}.
+ * On-heap buffer for values that provides a sorted view of itself as a {@link SegmentRamIterator}.
  */
 @NotThreadSafe
-public class BlockBalancedTreeRamBuffer
+public class SegmentRamBuffer
```

Comment:
```suggestion
public class TrieSegmentBuffer
```
We might as well use "trie" to be a little more descriptive. The trie is essential rather than accidental.

Reply:
I would prefer `SegmentTrieBuffer`, but I agree that it should have trie in the name somewhere.

Reply:
Yeah, my thought was it's a `SegmentBuffer` (imagine a future interface) that uses a trie /shrug